L1Trigger/L1TZDC/plugins/L1TZDCProducer.cc

```diff
@@ -0,0 +1,4885 @@
+      continue;
+    if (section != 1 && section != 2)
+      continue;
+    if (section == 1 && (channel < 1 || channel > 5))
+      continue;
+    if (section == 2 && (channel < 1 || channel > 4))
+      continue;
+    int ihitid = (zside == 1 ? 9 : 0) + (section == 2 ? 5 : 0) + (channel - 1);
+    //std::cout<<"------ channel ------ "<<channel<<std::endl;
+    counter++;
+    //the loop below iterates over the time slices
+    for (int iTS = 0; iTS < nSamples; iTS++) {
+      //std::cout<<" ------ iTS= "<<iTS<<std::endl;
+      unsigned short adc = (unsigned short)frame[iTS].adc();
+      rawadc[ihitid][iTS] = adc;
+      //std::cout<<"ihitid= "<<ihitid<<" , iTS= "<<iTS<<", ADC= "<<adc<<", rawadc[ihitid][iTS] ="<<rawadc[ihitid][iTS]<<std::endl;
+    }  // end of loop over iTS
+  }    //end of loop over channels
+
+  //std::cout<<"my counter= "<<counter<<std::endl;
+  for (int ibx = 0; ibx < nSamples; ibx++) {
+    double cEMP = 0, cEMM = 0, cHDP = 0, cHDM = 0;
+    double sumcEMP = 0, sumcEMM = 0, sumcHDP = 0, sumcHDM = 0;
+    //idet=0-4 correpond to the EM channels
+    for (int idet = 0; idet < 5; idet++) {
+      unsigned short EMP = rawadc[idet + 9][ibx];
+      unsigned short EMM = rawadc[idet][ibx];
+
+      if (!m_doHardCodeLUT) {
+        int cEMP_LUTIndex = zdcLUTIndexHelper(idet + 9, (int)EMP);
+        int cEMM_LUTIndex = zdcLUTIndexHelper(idet, (int)EMM);
+
+        cEMP = ((double)m_params->zdcLUT()->data(cEMP_LUTIndex)) / ((double)m_scaleFactor);
+        cEMM = ((double)m_params->zdcLUT()->data(cEMM_LUTIndex)) / ((double)m_scaleFactor);
+      }  //Old hardcoded lut from @ginnocen - superseded by above
+      else {
+        cEMP = QIE10_regular_fC_full[(UChar_t)(EMP)][idet + 9];
+        cEMM = QIE10_regular_fC_full[(UChar_t)(EMM)][idet];
+      }
+
+      sumcEMP = sumcEMP + cEMP;
+      sumcEMM = sumcEMM + cEMM;
+    }
+    //idet=5-8 correspond to HAD channels
+    for (int idet = 5; idet < 9; idet++) {
+      unsigned short HDP = rawadc[idet + 9][ibx];
+      unsigned short HDM = rawadc[idet][ibx];
+
+      if (!m_doHardCodeLUT) {
+        int cHDP_LUTIndex = zdcLUTIndexHelper(idet + 9, (int)HDP);
+        int cHDM_LUTIndex = zdcLUTIndexHelper(idet, (int)HDM);
+
+        cHDP = ((double)m_params->zdcLUT()->data(cHDP_LUTIndex)) / ((double)m_scaleFactor);
+        cHDM = ((double)m_params->zdcLUT()->data(cHDM_LUTIndex)) / ((double)m_scaleFactor);
+      }  //Old hardcoded lut from @ginnocen - superseded by above
+      else {
+        cHDP = QIE10_regular_fC_full[(UChar_t)(HDP)][idet + 9];
+        cHDM = QIE10_regular_fC_full[(UChar_t)(HDM)][idet];
+      }
+
+      sumcHDP = sumcHDP + cHDP;
+      sumcHDM = sumcHDM + cHDM;
+    }
+    double sumM = sumcEMM + sumcHDM;
+    double sumP = sumcEMP + sumcHDP;
+    //std::cout<<"bx = "<<ibx<<", sumP= "<<sumP<<", sumM= "<<sumM<<std::endl;
+    if (ibx == 4) {
+      std::cout << ", sumM= " << sumM << std::endl;
+      std::cout << ", sumP= " << sumP << std::endl;
```

Comment:
`std::cout` should probably be avoided here in favor of `edm::LogInfo`, `edm::LogDebug`, or `edm::LogWarning`, whichever is more appropriate when this kind of information is needed.

Reply:
Opted for LogInfo - in latest commit